filter/big_int.go

```diff
@@ -0,0 +1,76 @@
+package filter
+
+import (
+	bigFilter "github.com/silkeh/euler/filter/big"
+	"math/big"
+	"sync"
+)
+
+// BigInt defines a Filter that can handle big.Int.
+type BigInt struct {
+	filters []bigFilter.Filter
+}
+
+// NewBigInt returns a Filter that internally works with *big.Int, allowing values bigger than 2^64.
+// toBig defines how the int input is changed to *big.Int.
+// toFilter defines how the *big.Int is converted to int.
+// bigFilters defines the list of BigFilters where the *big.Int is parsed through
+func NewBigInt(bigFilters ...bigFilter.Filter) Filter {
+	return &BigInt{
+		filters: bigFilters,
+	}
+}
+
+// Run against an input and output channel until the input channel is closed.
+// The output channel is closed on return.
+func (bi *BigInt) Run(in <-chan int, out chan<- int) {
+	var wg sync.WaitGroup
+	wg.Add(len(bi.filters) + 2)
+
+	defer func() {
+		wg.Wait()
+	}()
```

Comment:
```
defer wg.Wait()
```

Reply:
Also, this can happen at the end of the function, as there are no other return paths.